react-textarea: Updating spec to match implementation and adding upgrade guide. (#22902)
microsoft/fluentui: apps/public-docsite-v9/src/Concepts/Upgrade/FromV8/Components/Textarea.stories.mdx

import { Meta } from '@storybook/addon-docs';

<Meta title="Concepts/Upgrading/from v0/Components/Textarea Upgrade" />

# Textarea Upgrade

> Fluent UI v8 provides the `TextField` control to allow users to enter and edit text. Fluen UI v9 provides a `Textarea` control, but has a different API.

Reviewer: _nit:_

```suggestion
Fluent UI v8 provides the `TextField` control to allow users to enter and edit text. Fluent UI v9 provides a `Textarea` control, but has a different API.
```

> v9 has takes a different approach on the `TextField` component and uses the `Input` and `Textarea` components. `Input` in this case is the general `TextField` component and `Textarea` is a `TextField` with the `multiline` prop set to `true`.

Reviewer: _nit:_ wording

```suggestion
v9 takes a different approach than v8 with respect to text inputs, and instead of using a single `TextField` component instaed uses separate `Input` and `Textarea` components. v9's `Input` in this case is equivalent to the general v8's `TextField`, while v9's `Textarea` is equivalent to v8's `TextField` with the `multiline` prop set to `true`.
```

## Examples

### Basic Upgrade

Basic usage of `TextField` v8

```tsx
import React from 'react';
import { TextField } from '@fluentui/react/lib/TextField';

const TextAreaV0BasicExample = () => {
  return (
    <>
      <TextField label="Standard" multiline rows={3} />
      <TextField label="Disabled" multiline rows={3} disabled defaultValue={dummyText} />
      <TextField label="Non-resizable" multiline resizable={false} />
    </>
  );
};

export default TextAreaExample;
```

An equivalent `Textarea` in v9 is

```tsx
import * as React from 'react';
import { Label } from '@fluentui/react-components';
import { Textarea } from '@fluentui/react-components/unstable';
import { useId } from '@fluentui/react-utilities';

const TextareaV9BasicExample = () => {
  const standardTextareaId = useId('standard');
  const disabledTextareaId = useId('disabled');
  const nonResizableTextareaId = useId('nonResizable');

  return (
    <>
      <div>
        <Label htmlFor={standardTextareaId}>Standard</Label>
        <Textarea id={standardTextareaId} />
      </div>
      <div>

>         <Label disabled defaultValue={dummyText} htmlFor={disabledTextareaId}>
>           Disabled
>         </Label>
>         <Textarea disabled id={disabledTextareaId} />

Reviewer: ```suggestion
        <Label disabled htmlFor={disabledTextareaId}>
          Disabled
        </Label>
        <Textarea defaultValue={dummyText} disabled id={disabledTextareaId} />
```

| `styles`                   | `className`    |                                                                                                                  |
| `suffix`                   | n/a            |                                                                                                                  |
| `theme`                    | n/a            | use `FluentProvider` to customize themes                                                                         |
| `underlined`               | n/a            |                                                                                                                  |
| `validateOnFocusIn`        | n/a            |                                                                                                                  |
| `validateOnFocusOut`       | n/a            |                                                                                                                  |
| `validateOnLoad`           | n/a            |                                                                                                                  |

> | `value`                    | n/a            | Mutually exclusive with `defaultValue`                                                                           |

Reviewer: ```suggestion
| `value`                    | `value`        | Mutually exclusive with `defaultValue`                                                                           |
```
